### PR TITLE
Add multi input/output channels support to migraphx worker; new example scripts for Yolo and Bert models"

### PR DESCRIPTION
**Pull request by bpickrel**
# Summary of Changes

*  Added support for multiple input or output channels, needed by many current machine learning models.
*  Added example scripts for Yolo object detection (1 input; 3 outputs) and Bert language processing (4 inputs; 3 outputs) which could not be run before this.
*  Modified existing hello-migraphx example to submit multiple requests asynchronously, to comply with the engine's recent change in batching interface.

### CODE REVIEW

File: examples/python/migraphx_bert.py [hunk 1]
```diff
@@ -0,0 +1,260 @@
+# Copyright 2022 Advanced Micro Devices, Inc.
```
Comment (varunsh-xilinx) [inline examples/python/migraphx_bert.py:1]: Why are there two sets of licenses in this file? If you've copied this from somewhere, keep the original license

File: examples/python/migraphx_bert.py [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+wget -q https://storage.googleapis.com/bert_models/2018_10_18/uncased_L-12_H-768_A-12.zip
+unzip uncased_L-12_H-768_A-12.zip
+rm uncased_L-12_H-768_A-12.zip
+
+[in Docker]   pip3 install tokenizers
+
+[in bash ]
+cp /home/bpickrel/is-fork/inference-server/AMDMIGraphX/examples/nlp/python_bert_squad/requirements_bertsquad.txt  ~/is-fork/inference-server/external/artifacts/migraphx/bert_squad/
+cp /home/bpickrel/is-fork/inference-server/AMDMIGraphX/examples/nlp/python_bert_squad/run_onnx_squad.py ~/is-fork/inference-server/external/artifacts/migraphx/bert_squad/
+cp /home/bpickrel/is-fork/inference-server/AMDMIGraphX/examples/nlp/python_bert_squad/*.json  ~/is-fork/inference-server/external/artifacts/migraphx/bert_squad/
```
Comment (varunsh-xilinx) [inline examples/python/migraphx_bert.py:64]: Delete these notes for yourself

**Review by varunsh-xilinx — COMMENTED** [review]
What is the source for the crowd.jpg image?

**Comment by bpickrel** [comment]
> What is the source for the crowd.jpg image?

https://farm3.staticflickr.com/2009/2306189268_88cc86b30f_z.jpg and the MIGraphX repo

File: examples/python/migraphx_bert.py (hunk 2 above)
Comment (bpickrel) [reply examples/python/migraphx_bert.py:64]: done

File: examples/python/migraphx_bert.py (hunk 1 above)
Comment (bpickrel) [reply examples/python/migraphx_bert.py:1]: inserted MIT license (only) in all examples

**Comment by varunsh-xilinx** [comment]
Why have you changed all of them to MIT now? If there's a new file you're adding that you've largely copied from somewhere else and it comes with an existing license header, then you should keep that header but for your own work, continue with Apache 2. Your existing examples shouldn't be relicensed

**Comment by varunsh-xilinx** [comment]
> > What is the source for the crowd.jpg image?
> 
> https://farm3.staticflickr.com/2009/2306189268_88cc86b30f_z.jpg and the MIGraphX repo

Thanks. The image is from https://www.flickr.com/photos/mattmangum/2306189268/ under a CC BY 2.0 license. Is there a reason you're adding this file for the yolo example rather than using the existing bicycle picture? That image gives you nicer boxes with higher confidences anyway

**Comment by bpickrel** [comment]
> 

I got the link to the picture from the AMDMIGraphX repository.  I'm using it because it's the same image as the script I modeled mine on, and I can compare results.  If I add the image to the "proteus get" downloads to avoid the attribution problems of putting it in the repo, does it matter what directory it goes in?

**Comment by bpickrel** [comment]
> 

This is per instructions from MVermeulen and guidance from company legal.
The scripts they're based on come from our team, and we've had an initiative in recent months to correct the license statements on all our files.  The MIT licenses are the ones from the source scripts.

**Comment by varunsh-xilinx** [comment]
Okay. I think the image is fine for now. Since my other branch is changing the downloading script anyway, I'll remove this file from the repository in that branch and download it with `proteus get` instead.

**Comment by varunsh-xilinx** [comment]
Is this okay to merge or are you still adding more commits?

**Comment by bpickrel** [comment]
> 

I'm finally done.

**Comment by bpickrel** [comment]
Closed by mistake; clicked the wrong button.  Reopened!

**Pull request merged**